monitoring_dashboard.qmd

```diff
@@ -0,0 +1,138 @@
+---
+title: "Monitoring dashboard"
+format:
+  dashboard:
+    orientation: columns
+logo: https://github.com/rstudio/vetiver-python/blob/main/docs/figures/logo.png?raw=true
+output: asis
+jupyter: python3
+---
+
+```{python}
+#| include: false
+#| tags: [parameters]
+
+# import model and metadata
+import pins
+from IPython.display import display, Markdown, IFrame
+from datetime import datetime, timedelta
+import pandas as pd
+import plotly.express as px
+from sklearn import metrics
+from vetiver import VetiverModel, compute_metrics, plot_metrics
+from sklearn.metrics import recall_score, accuracy_score
+
+raw = "https://colorado.rstudio.com/rsc"
+paths = {"chicago-model-python": "chicago-model-python/"}
+board = pins.board_url(raw, paths, allow_pickle_read=True)
+v = VetiverModel.from_pin(board, "chicago-model-python")
+v_meta = board.pin_meta("chicago-model-python")
+days_old = datetime.today() - datetime.strptime(v_meta.created, "%Y%m%dT%H%M%SZ")
+```
+
+```{python}
+## the next few lines are an example model, here is a place to
+## add any code you need to import new data and make predictions
+
+# import new data to track performance over time
+raw = "https://colorado.rstudio.com/rsc"
+paths = {"new-data": "inspections-new-data/"}
+board = pins.board_url(raw, paths, allow_pickle_read=True)
+inspections_new = board.pin_read("new-data")
+
+# make predictions
+inspections_new["preds"] = v.model.predict(
+    inspections_new.drop(columns=["results", "aka_name", "inspection_date"])
+)
+
+# map results
+inspections_new["preds"] = inspections_new["preds"].map({"PASS": 0, "FAIL": 1})
+inspections_new["results"] = inspections_new["results"].map({"PASS": 0, "FAIL": 1})
+```
+
+# Model card
+## Column
+### Row {height="33%"}
+::: {.valuebox}
+`{python} v.description`
+
+`{python} v.model_name`
+:::
+
+::: {.valuebox}
+Model age
+
+`{python} days_old.days` days old
+:::
+
+### Row
+
+Model details
+
+- The data used ...
+- The model was created by ...
+
+Intended use
+
+- The primary intended users of this model ...
+- Some use cases are out of scope for this model ...
+
+Training data & evaluation data
+
+- The training dataset for this model has the prototype:
+```
+`{python} v.prototype.construct().schema().get("properties")`
+```
+
+Ethical considerations
+
+- It was considered ...
+
+Caveats & recommendations
+
+- We recommend ...
+
+# Explore validation data
```

Comment:
```suggestion
# Model metrics
```

Or "Model performance" or similar. I think "Explore validation data" implies more, say, EDA of the validation data vs. connecting back to the model.

Showing an example of that (just a histogram for the validation data) is included in the R template, and I do think that's a useful category of thing to show people (not metrics, just what the new data is like). Thoughts?

Reply:
Ah, sure! That makes sense to encourage more exploration of incoming data. I'll add that in.